lib/services/volunteer.js

```diff
@@ -2,137 +2,80 @@ const Schmervice = require('schmervice');
 const { errorHandler } = require('../errorHandling');
 
 module.exports = class UserRoleService extends Schmervice.Service {
-  static calculateEngagement(d1, d2) {
-    d1 = new Date(d1);
-    d2 = new Date(d2);
-    let months;
-    months = (d2.getFullYear() - d1.getFullYear()) * 12;
-    months -= d1.getMonth();
-    months += d2.getMonth();
-    if (months <= 0) {
-      return 0;
-    }
-    return months;
-  }
-
   async getAllVolunteers() {
-    const { UserRole, User, Classes } = this.server.models();
-    const volunteers_data = {};
+    const { UserRole, Classes } = this.server.models();
+    let volunteers = [];
 
     try {
-      // get user_ids from userRole
-      const volunteer_ids = await UserRole.query()
+      // get volunteer data and partner data
+      volunteers = await UserRole.query()
         .select('user_id as id')
-        .where('role', 'volunteer');
+        .where('role', 'volunteer')
+        .withGraphFetched('user.[partner]');
 
-      /**
-       * You should use withGraphFetched function of Objection to leverage the relation mapping of UserRole model
-       * with Users model further with classes model
-       *
-       * Something like this
-       * const volunteers = await UserRole.query().where('role', 'volunteer').withGraphFetched('user');
-       *
-       * Just take care of one thing Classes -> User relation mapping is done on keys classes.user_id -> users.id, so this will give you the data
-       * of students joining the class and not the facilitators
-       *
-       */
-
-      volunteer_ids.forEach((v, index) => {
-        volunteer_ids[index] = v.id;
+      volunteers.forEach((v, index) => {
+        volunteers[index] = volunteers[index].user;
       });
```

Comment:
Can also use the following (just be sure to wrap the result of `await` in parentheses so you're working with the `Array`):
```suggestion
      volunteers = (await UserRole.query()
        .select('user_id as id')
        .where('role', 'volunteer');
        .where('role', 'volunteer')
        .withGraphFetched('user.[partner]'))
        .map(v => v.user);
```

Reply:
Thanks @jschanker :)

Reply:
@AnandDhakane01 Please look into these suggestions. I cannot merge @jschanker 's commit directly because the code is outdated and some lines are already deleted. However, I will make these changes and push the commits

Reply:
@pysaquib Happy to support the great work you all do in the little ways I can!